src/main/java/io/pinecone/commons/IndexInterface.java

```diff
@@ -18,11 +21,46 @@ default UpsertRequest validateUpsertRequest(String id,
                                                 String namespace) {
         UpsertRequest.Builder upsertRequest = UpsertRequest.newBuilder();
 
+        Vector vector = buildUpsertVector(id, values, sparseIndices, sparseValues, metadata);
+
+        upsertRequest.addVectors(vector);
+
+        if (namespace != null) {
+            upsertRequest.setNamespace(namespace);
+        }
+        return upsertRequest.build();
+    }
+
+    default UpsertRequest validateUpsertRequest(List<VectorWithUnsignedIndices> vectorWithUnsignedIndicesList,
```

Comment:
Added upsert(List<Vectors>, String namespace) to mimic Node and python sdk upserts